src/symbolize/gimli.rs

```diff
@@ -329,28 +329,35 @@ impl Mapping {
     }
 }
 
-thread_local! {
-    // A very small, very simple LRU cache for debug info mappings.
-    //
-    // The hit rate should be very high, since the typical stack doesn't cross
-    // between many shared libraries.
-    //
-    // The `addr2line::Context` structures are pretty expensive to create. Its
-    // cost is expected to be amortized by subsequent `locate` queries, which
-    // leverage the structures built when constructing `addr2line::Context`s to
-    // get nice speedups. If we didn't have this cache, that amortization would
-    // never happen, and symbolicating backtraces would be ssssllllooooowwww.
-    static MAPPINGS_CACHE: RefCell<Vec<(PathBuf, Mapping)>>
-        = RefCell::new(Vec::with_capacity(MAPPINGS_CACHE_SIZE));
+// A very small, very simple LRU cache for debug info mappings.
+//
+// The hit rate should be very high, since the typical stack doesn't cross
+// between many shared libraries.
+//
+// The `addr2line::Context` structures are pretty expensive to create. Its
+// cost is expected to be amortized by subsequent `locate` queries, which
+// leverage the structures built when constructing `addr2line::Context`s to
+// get nice speedups. If we didn't have this cache, that amortization would
+// never happen, and symbolicating backtraces would be ssssllllooooowwww.
+static mut MAPPINGS_CACHE: Option<Mutex<Vec<(PathBuf, Mapping)>>> = None;
+
+fn lazy_cache() -> &'static Mutex<Vec<(PathBuf, Mapping)>> {
```

Comment:
If you're gonna go ahead and drop the `thread_local!` then the `Mutex` here can be dropped as well. All of the function in this module are externally synchronized so it's ok to to use the `static mut` here. (FWIW this implementation isn't threadsafe since `get_or_insert_with` isn't threadsafe, but it's ok in this specific context since thread safety isn't necessary).

Could this also be switched to be the same interface as before, something like:

```rust
fn with_cache(f: impl FnOnce(&mut Vec<(PathBuf, Mapping)>)) {
    // ...
}
```

so that way the `'static` lifetime won't leak

Reply:
You're right, I didn't realize that once the `unsafe` was there then the `Mutex` wasn't needed anymore. And the `with_cache` function can wrap the `static` entirely, so it's not visible outside, which is nice.